ui/apps/platform/cypress/integration/violations/violationsSearchFilter.test.ts

```diff
@@ -0,0 +1,125 @@
+import withAuth from '../../helpers/basicAuth';
+import { selectFilteredWorkflowView, visitViolations } from './Violations.helpers';
+
+function selectCompoundSearchFilterAttribute(attribute: string) {
+    cy.get('button[aria-label="compound search filter attribute selector toggle"]').click();
+    cy.get(
+        `ul[aria-label="compound search filter attribute selector menu"] button:contains("${attribute}")`
+    ).click();
+}
+
+function selectFirstAutocompleteSearchOption() {
+    cy.get(
+        'div[aria-labelledby="Filter results menu toggle"] button[aria-label="Menu toggle"]'
+    ).click();
+    cy.get(`ul[aria-label="Filter results select menu"] li:nth(0) button`).click();
+}
+
+describe('Violations - Search Filter', () => {
+    withAuth();
+
+    beforeEach(() => {
+        // clear the filters before every test
+        cy.document().then((doc) => {
+            const button = Cypress.$('button:contains("Clear filters")', doc);
+            if (button.length > 0) {
+                // if button exists, click it
+                cy.wrap(button).click();
+            }
+        });
```

Comment:
Two things come to mind with this:

1. In what situation are filters applied when visiting the Violations page? I don't see this behavior locally.
2. Is it possible that this always has a result of `button.length === 0`? It looks like the query for the buttons will occur as soon as the document is ready which could be before React has had a chance to render that button.

Reply:
I'm afraid I can't remember why I added that 🤔 but yeah it does look unnecessary now. I will remove that.